lib/opal/cli_runners/chrome_cdp_interface.rb

```diff
@@ -3,143 +3,185 @@
 # This script I converted into Opal, so that I don't have to write
 # buffer handling again. We have gets, Node has nothing close to it,
 # even async.
+# For CDP see docs/cdp_common.(md|json)
 
 require 'opal/platform'
 
 %x{
 var CDP = require("chrome-remote-interface");
 var fs = require("fs");
+var http = require("http");
 
 var dir = #{ARGV.last}
+var offset; // port offset for http server, depending on number of targets
 
 var options = {
   host: #{ENV['CHROME_HOST'] || 'localhost'},
   port: #{ENV['CHROME_PORT'] || 9222}
 };
 
+function not_found(res) {
+  res.writeHead(400, { "Content-Type": "text/plain" });
+  res.end("NOT FOUND");
+}
+
+var server = http.createServer(function(req, res) {
```

Comment:
since chromium runner can work with a server, perhaps we can collapse the two cdp interfaces?

Reply:
Probably we can support more features in the future (coverage, performance, profiling, debugging), however, thats where FF and Chrome start to differ even more, thats why i like to keep them separate. They already differ slightly, they just appear similar (eg. stack representation).